src/rgw/rgw_op.cc

```diff
@@ -2895,10 +3020,20 @@ void RGWPostObj::execute()
 
   processor = select_processor(*static_cast<RGWObjectCtx *>(s->obj_ctx));
 
+  // no filters by default
+  filter = processor;
+
   op_ret = processor->prepare(store, NULL);
   if (op_ret < 0)
     goto done;
 
+  compression_enabled = s->cct->_conf->rgw_compression_type != "none";
+  if (compression_enabled) {
+    filter = new RGWPutObj_Compress(s->cct, filter);
+    // save, because filter can be rewritten later
+    compressor = (RGWPutObj_Compress*)filter;
+  }
+
```

Comment:
similar to `RGWGetObj`, consider using `boost::optional` instead of heap allocation